subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -264,13 +264,16 @@ void pe_report_error(const struct device *dev, const enum pe_error e,
 	 *     response.
 	 */
 	/* All error types besides transmit errors are Protocol Errors. */
-	if ((e != ERR_XMIT && atomic_test_bit(pe->flags, PE_FLAGS_INTERRUPTIBLE_AMS) == false) ||
-	    e == ERR_XMIT ||
+	if ((e != ERR_XMIT) && ((atomic_test_bit(pe->flags, PE_FLAGS_INTERRUPTIBLE_AMS) == false) ||
 	    (atomic_test_bit(pe->flags, PE_FLAGS_EXPLICIT_CONTRACT) == false &&
-	     type == PD_PACKET_SOP)) {
+	     type == PD_PACKET_SOP))) {
 		policy_notify(dev, PROTOCOL_ERROR);
 		pe_send_soft_reset(dev, type);
 	}
+	/* Transmit error */
+	else if (e == ERR_XMIT) {
```

Comment:
Done

Reply:
You probably can just drop this commit, because you're refactoring and fixing the soft reset handling [here](https://github.com/zephyrproject-rtos/zephyr/pull/52993/commits/a616f11e92a0ef7f476f0c7af77f692f85d7562e#)

As is, this commit has an error regarding the non-AMS check - you also need to qualify that the message is an SOP.

Reply:
This commit adds the PE_FLAGS_MSG_XMIT_ERROR flag, so let's keep this commit instead of moving the flag to another commit.